### ISSUE TITLE
Doesn't inject sources after v4

### ISSUE THREAD

**Issue by atilkan** [jantimon/html-webpack-plugin#1381]
This is my setting. It doesn't inject sources. It doesn't matter if i inject manually using ejs or using inject true.
```
template: path.join(`${page.file}.ejs`),
filename: path.join(`${page.file}.html`),
chunks: [page.key],
inject: true,
minify: false,
```

If i just change the version to 3.0.2 it works. 
Something is fucked up again. 

Windows 10.

**Comment by jantimon** [comment]
Can you please try InlineChunkHtmlPlugin from facebook? the https://github.com/facebook/create-react-app/blob/edc671eeea6b7d26ac3f1eb2050e50f75cf9ad5d/packages/react-dev-utils/InlineChunkHtmlPlugin.js#L10

**Comment by cdagli** [comment]
Same for me, our configuration stopped working after updating v4.x.x. We've reverted back to 3.2.0 and investigating if something wrong with our configuration. 

We don't have anything magical, it is just straightforward configuration:

```
    new HtmlWebpackPlugin({
      template: './src/index.ejs',
    }),
```

**Comment by jantimon** [comment]
@cdagli the issue you mentioned is not related - it is about output filenames

could you please show how you use the `css` / `js`  inside your index.ejs?

**Comment by odrozd** [comment]
@jantimon having similar issue after upgrading to v4 and working fine in v3
here is config:
```
 new HtmlWebpackPlugin({
            filename: "../Areas/MVC/Views/WebPackBundles/_ConnectionLoggingBundles.cshtml",
            chunks: ['manifest', 'vendor', 'connection_logging_loader'],
            inject: false,
            template: '_LayoutTemplate.cshtml'
        })
```
and template
```
<% for (var chunk in htmlWebpackPlugin.files.chunks) { %>
<script src="~/<%= htmlWebpackPlugin.files.chunks[chunk].entry %>"></script>
<% } %>
```

**Comment by cdagli** [comment]
@jantimon Sorry, maybe I got this wrong and our issue is unrelated with this issue but related with #1355 Because it works when we remove the `?[chunkhash]` in the output.filename `[name].js?[chunkhash]'`

**Comment by odrozd** [comment]
just fyi removing `[chunkhash]` does not work in my case

**Comment by kroko** [comment]
Doing

```ejs
  <!--
    // DEBUG
    <%= JSON.stringify(htmlWebpackPlugin.files, null, 1) %>
  -->
```

in template will reveal structure such as this

```
<!--
    // DEBUG
    {
 "publicPath": "assets/",
 "js": [
  "assets/runtime.8379fd9de1396362f7d6.js",
  "assets/vendors.63498add9423ff0ff9a0.js",
  "assets/corepoly.b51addbe29703dc78ebb.js",
  "assets/index.9eca6c1781f27940f1de.js"
 ],
 "css": [
  "assets/index.5728905c20b76099151d.css"
 ]
}
  -->
```

thus not the one as shown in docs https://github.com/jantimon/html-webpack-plugin#writing-your-own-templates

```
"htmlWebpackPlugin": {
  "files": {
    "css": [ "main.css" ],
    "js": [ "assets/head_bundle.js", "assets/main_bundle.js"],
    "chunks": {
      "head": {
        "entry": "assets/head_bundle.js",
        "css": [ "main.css" ]
      },
      "main": {
        "entry": "assets/main_bundle.js",
        "css": []
      },
    }
  }
}
```

Thus template that worked with 3.2.0

```ejs
<% for (let key in htmlWebpackPlugin.files.chunks) { %>
   <script src="<%= htmlWebpackPlugin.files.chunks[key].entry %>"></script>
<% } %>
```

does not work any more.

**Comment by kroko** [comment]
oh, is see, it is already discussed here https://github.com/jantimon/html-webpack-plugin/issues/1369

**Comment by jantimon** [comment]
The querystring was solved with html-webpack-plugin 4.0.4

In html-webpack-plugin `<%= htmlWebpackPlugin.files` is no longer supported.

More details why that change was made can be found here: https://github.com/jantimon/html-webpack-plugin/issues/1369#issuecomment-604382645

I will try to provide a new way but it will take me some time as I can do that only in my spare time.

So unfortunately for now you have to use html-webpack-plugin 3.2 for now if you rely on `<%= htmlWebpackPlugin.files`.

**Comment by jantimon** [comment]
Closing as it is a duplicate of #1369
